opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -471,6 +473,22 @@ def detect(self) -> "Resource":
         )
 
 
+class _HostResourceDetector(ResourceDetector):
+    """
+    The HostResourceDetector detects the hostname and architecture attributes.
+    """
+
+    def detect(self) -> "Resource":
+        _host_name = socket.gethostname()
+        _host_arch = platform.machine()
+        return Resource(
+            {
+                HOST_NAME: _host_name,
```

Comment:
Maybe something like this instead?
```suggestion
                HOST_NAME: socket.gethostname(),
```